From 52b84c8693d99fe877d9e612881503fe28c39210 Mon Sep 17 00:00:00 2001
From: Sven Joachim <svenjoac@gmx.de>
Date: Fri, 28 Apr 2017 09:23:20 +0200
Subject: [PATCH] Handle case where some other awk than mawk is required

Set up original-awk or gawk, if they had been downloaded instead of
mawk.  This should make it possible to let either of those have
"Priority: required" and downgrade mawk's priority.
---
 scripts/debian-common | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/scripts/debian-common b/scripts/debian-common
index 211a4c4..88dbc07 100644
--- a/scripts/debian-common
+++ b/scripts/debian-common
@@ -9,6 +9,13 @@ case $ARCH in
 	*)          LIBC="libc6" ;;
 esac
 
+find_awk () {
+	if [ -x "$TARGET/usr/bin/mawk" ]; then AWK=mawk
+	elif [ -x "$TARGET/usr/bin/original-awk" ]; then AWK=original-awk
+	elif [ -x "$TARGET/usr/bin/gawk" ]; then AWK=gawk
+	fi
+}
+
 work_out_debs () {
 	required="$(get_debs Priority: required)"
 
@@ -117,7 +124,8 @@ Status: install ok installed" >> "$TARGET/var/lib/dpkg/status"
 	info INSTCORE "Installing core packages..."
 
 	p; progress $baseprog $bases INSTCORE "Installing core packages" #2
-	ln -sf mawk "$TARGET/usr/bin/awk"
+	find_awk
+	[ -n "$AWK" ] && ln -sf $AWK "$TARGET/usr/bin/awk"
 	x_core_install base-passwd
 	x_core_install base-files
 	p; progress $baseprog $bases INSTCORE "Installing core packages" #3
@@ -138,8 +146,8 @@ Status: install ok installed" >> "$TARGET/var/lib/dpkg/status"
 	x_core_install perl-base
 
 	p; progress $baseprog $bases INSTCORE "Installing core packages" #6
-	rm "$TARGET/usr/bin/awk"
-	x_core_install mawk
+	rm -f "$TARGET/usr/bin/awk"
+	[ -n "$AWK" ] && x_core_install $AWK
 
 	p; progress $baseprog $bases INSTCORE "Installing core packages" #7
 	if doing_variant -; then
-- 
2.18.0

